src/Native/System.Security.Cryptography.Native/pal_ssl.cpp

```diff
@@ -44,9 +57,36 @@ extern "C" SSL_CTX* SslCtxCreate(SSL_METHOD* method)
     return SSL_CTX_new(method);
 }
 
-extern "C" int64_t SslCtxCtrl(SSL_CTX* ctx, int32_t cmd, int64_t larg, void* parg)
+extern "C" void SetProtocolOptions(SSL_CTX* ctx, SslProtocols protocols)
 {
-    return SSL_CTX_ctrl(ctx, cmd, larg, parg);
+    long protocolOptions = 0;
+
+    if ((protocols & PAL_SSL_Ssl2) != PAL_SSL_Ssl2)
+    {
+        protocolOptions |= SSL_OP_NO_SSLv2;
+    }
+    if ((protocols & PAL_SSL_Ssl3) != PAL_SSL_Ssl3)
+    {
+        protocolOptions |= SSL_OP_NO_SSLv3;
+    }
+    if ((protocols & PAL_SSL_Tls) != PAL_SSL_Tls)
+    {
+        protocolOptions |= SSL_OP_NO_TLSv1;
+    }
+#if HAVE_TLS_V1_1
```

Comment:
We will get angry before this:
https://github.com/dotnet/corefx/blob/master/src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs#L266

All this code is doing is flipping off these options if the protocol isn't set in our flags enum.  So if the caller says they want SslProtocols.Ssl2 | SslProtocols.Tls11, there is no reason to get angry here if TLS1.1 isn't available. We will end up using Ssl2 according to that GetSslMethod.

Looking at that method, I am wondering if it is implemented correctly... if all TLS flags are set, it looks like it will use Ssl2.  Is that really want we want?